### PR TITLE
Debugging support

### PR DESCRIPTION
**Pull request by edfelten**
When EVM code does a REVERT instruction, emit a debugprint of `(10000, contractAddress, evmPC)` where `contractAddress` is the L2 address of the contract that did the revert, and `evmPC` is the EVM program counter where the revert occurred.

### CODE REVIEW

File: arb_os/arbsys.mini [hunk 1]
```diff
@@ -65,12 +67,14 @@ import impure func decompressAddress(stream: ByteStream) -> option<(ByteStream,
...
 
+import impure func pcTableForCode(code: ByteArray) -> option<Stack>;
+
 
 public impure func arbsys_txcall() {
     if let Some(topFrame) = evmCallStack_topFrame() {
         let calldata = evmCallFrame_getCalldata(topFrame);
         if (bytearray_size(calldata) < 4) {
-            evmOp_revert(0, 0);
+            evmOp_revert_knownCodePc(address(100), 0, 0, 0);
```
Comment (hkalodner) [inline arb_os/arbsys.mini:77]: It might be better for debugging to give each ArbSys revert it's own artificial PC so that they can be distinguished.

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: arb_os/arbsys.mini (hunk 1 above)
Comment (edfelten) [reply arb_os/arbsys.mini:79]: This is done in the next commit.

File: arb_os/arbsys.mini [hunk 2]
```diff
@@ -65,12 +69,14 @@ import impure func decompressAddress(stream: ByteStream) -> option<(ByteStream,
...
```
Comment (hkalodner) [inline arb_os/arbsys.mini:79]: It looks like `evmOp_revert_knownCodePc(address(100), 0, 0, 0);` is still used a few times in this file. I'm not sure if that was intentional after the last commit replaced most of them

File: arb_os/codeSegment.mini [hunk 3]
```diff
@@ -516,8 +517,8 @@ public impure func translateEvmCodeSegment(
                     if (opcode == 0xfd) { // REVERT
                         return pushEvmInsnCall(opWrapper2Uint(evmOp_revert), restOfCode, evmJumpTable, pcTable);
```
Comment (hkalodner) [inline arb_os/codeSegment.mini:518]: Should this also be replaced with `evmOp_revert_knownPc`?

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: arb_os/codeSegment.mini (hunk 3 above)
Comment (edfelten) [reply arb_os/codeSegment.mini:518]: Fixed.

File: arb_os/arbsys.mini (hunk 2 above)
Comment (edfelten) [reply arb_os/arbsys.mini:79]: Fixed.

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**